view/frontend/web/internals/common.js

```diff
@@ -1,3 +1,54 @@
+var algolia = {
+	allowedHooks: [
+		'modifyAutocompleteSources',
+		'modifyAutocompleteOptions',
+		'algoliaHookBeforeInstantsearchInit',
+		'algoliaHookBeforeWidgetInitialization',
+		'algoliaHookBeforeInstantsearchStart',
+		'algoliaHookAfterInstantsearchStart'
+	],
+	registeredHooks: [],
+	registerHook: function (hookName, callback) {
+		if (this.allowedHooks.indexOf(hookName) === -1) {
+			throw 'Hook "' + hookName + '" cannot be defined. Please use one of ' + this.allowedHooks.join(', ');
+		}
+		
+		if (this.registeredHooks[hookName] === undefined) {
+			this.registeredHooks[hookName] = [];
+		}
+		
+		this.registeredHooks[hookName].push(callback);
+	},
+	getRegisteredHooks: function(hookName) {
+		if (this.registeredHooks[hookName] === undefined) {
+			return [];
+		}
+		
+		return this.registeredHooks[hookName];
+	},
+	triggerHooks: function () {
+		var hookName = arguments[0];
```

Comment:
I'd do the transformation from arguments to array here. And I would use a splice to remove the first element https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice